src/main/java/uk/ac/cam/cl/dtg/segue/api/QuestionFacade.java

```diff
@@ -196,13 +199,26 @@ public Response answerQuestion(@Context final HttpServletRequest request,
                 }
             } else {
                 try {
+                    // Monitor misuse on a per-question per-anonymous user basis:
                     misuseMonitor.notifyEvent(((AnonymousUserDTO) currentUser).getSessionId() + "|" + questionId,
                             AnonQuestionAttemptMisuseHandler.class.toString());
                 } catch (SegueResourceMisuseException e) {
                     this.getLogManager().logEvent(currentUser, request, QUESTION_ATTEMPT_RATE_LIMITED, response.getEntity());
                     String message = "You have made too many attempts at this question part. Please log in or try again later.";
                     return SegueErrorResponse.getRateThrottledResponse(message);
                 }
+                try {
+                    // And monitor on a blanket per IP Address basis for non-logged in users.
+                    // If we see serious misuse, this could be moved to *before* the attempt validation and checking,
+                    // to save server load. Since this occurs after the anon user notify event, that will catch most
+                    // misuse and this will catch misuse ignoring cookies or with repeated new anon accounts.
+                    misuseMonitor.notifyEvent(RequestIPExtractor.getClientIpAddr(request),
```

Comment:
My only concern is how this could interact with NAT based proxies in schools... If we get our threasholds right it shouldn't matter but we should just be careful.